src/brevitas/export/shark/manager.py

```diff
@@ -0,0 +1,170 @@
+class SharkManager(BaseManager):
+    # handlers = [SharkWeightQuant, SharkActQuant, SharkActFloatQuant, SharkWeightFloatQuant]
+
+    handlers = [SharkActEqualization, SharkLinearQuant, SharkQuantSDPA]
+
+    def __init__(self, config=None):
+        super().__init__()
+        if config == None:
+            config = dict()
+        self.config = config
+
+    @classmethod
+    def set_export_mode(cls, model: Module, enabled: bool):
+        _set_layer_export_mode(model, enabled)
+
+    @classmethod
+    def set_export_handler(cls, module: Module, name: str, shared_dict: dict):
+        _set_layer_export_handler(cls, module)
+        if hasattr(module, 'export_handler') and module.export_handler is not None:
+            module.export_handler.layer_name = name
+            module.export_handler.shared_dict = shared_dict
+
+    def gguf_preprocess(self, model):
+        import sys
+
+        import gguf
+
+        from brevitas_examples.llm.gguf_export.convert import ModelBase
+        """Export the model to gguf format."""
+        output_type = gguf.LlamaFileType.ALL_F32
+
+        config = model.config
+
+        tmp_work_dir = Path('./tmp_dir')
+        config.save_pretrained(tmp_work_dir)
+
+        with torch.no_grad():
+            hparams = ModelBase.load_hparams(tmp_work_dir)
+            model_architecture = hparams["architectures"][0]
+            try:
+                model_class = ModelBase.from_model_architecture(model_architecture)
+            except NotImplementedError:
+                sys.exit(1)
+            model_class = ModelBase.from_model_architecture(model_architecture)
+            model_name = model.name_or_path.split('/')
+            if len(model_name[-1]) == 0:
+                model_name = model_name[-2]
+            else:
+                model_name = model_name[-1]
+
+            model_instance = model_class(
+                model,
+                dir_model=tmp_work_dir,
+                ftype=output_type,
+                fname_out=tmp_work_dir,
+                is_big_endian=False,
+                model_name=model_name,
+                split_max_tensors=False,
+                split_max_size=0,
+                dry_run=False,
+                small_first_shard=False)
+        return model_instance
+
+    def export(self, model, *model_args, **model_kwargs):
+        from brevitas_examples.llm.main import offload_model
+        from brevitas_examples.llm.main import remove_hooks
+
+        shared_dict = {}
+
+        for name, module in model.named_modules():
+            self.set_export_handler(module, name, shared_dict)
+        self.set_export_mode(model, enabled=True)
+
+        print("Forward starts")
+        offload_model(model)
+        with torch.no_grad():
+            model(*model_args, **model_kwargs)
+        remove_hooks(model)
+        print("Forward ends")
+
+        self.set_export_mode(model, enabled=False)
+
+        updated_theta = dict()
+        gguf_model = self.gguf_preprocess(model)
```

Comment:
Add comment that we might want to generalize and/or abstract the mapping functionality